staging/src/k8s.io/client-go/tools/leaderelection/resourcelock/configmaplock.go

```diff
@@ -45,10 +45,11 @@ type ConfigMapLock struct {
 func (cml *ConfigMapLock) Get(ctx context.Context) (*LeaderElectionRecord, []byte, error) {
 	var record LeaderElectionRecord
 	var err error
-	cml.cm, err = cml.Client.ConfigMaps(cml.ConfigMapMeta.Namespace).Get(ctx, cml.ConfigMapMeta.Name, metav1.GetOptions{})
+	cm, err := cml.Client.ConfigMaps(cml.ConfigMapMeta.Namespace).Get(ctx, cml.ConfigMapMeta.Name, metav1.GetOptions{})
```

Comment:
Need to provide a comment explaining how the usage in the higher layer will handle stale data left in the instance variable.  

I know of two paths, the normal update path and the release path.  The normal update path doesn't progress without a successful get (I think), but the release path can happen without a recently successful get.

The comment should be here and in the update function.  Also, the nil check in update appears incorrect, so you probably want to address that too.